controllers/topology/internal/extensions/patches/inline/json_patch_generator.go

```diff
@@ -0,0 +1,372 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package inline implements the inline JSON patch generator.
+package inline
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/valyala/fastjson"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/extensions/patches/api"
+	patchvariables "sigs.k8s.io/cluster-api/controllers/topology/internal/extensions/patches/variables"
+	"sigs.k8s.io/yaml"
+)
+
+// JSONPatchGenerator generates JSON patches for a GenerateRequest based on a ClusterClassPatch.
+type JSONPatchGenerator struct {
+	patch *clusterv1.ClusterClassPatch
+}
+
+// NewJSONPatchGenerator returns a JSONPatchGenerator from a given ClusterClassPatch object.
+func NewJSONPatchGenerator(patch *clusterv1.ClusterClassPatch) api.Generator {
```

Comment:
```suggestion
// New returns a new inline Generator from a given ClusterClassPatch object.
func New(patch *clusterv1.ClusterClassPatch) api.Generator {
```

Reply:
This function and the struct above are already in the `patches/inline` package, which seems already self-explanatory